test/solution-template-validation-tests/test/mainTemplateTests.js

```diff
@@ -30,6 +30,19 @@ function getErrorMessage(obj, file, message) {
 /** Tests for template files in a solution template */
 describe('template files - ', () => {
 
+    /** Tests for $schema property in template files */
+    describe('$schema property tests - ', () => {
+        /** schema property MUST use https  */
+        it('schema property must use https not http', () => {
+            templateFileJSONObjects.forEach(templateJSONObject => {
+                var templateObject = templateJSONObject.value;
+                if (templateObject.$schema) {
```

Comment:
Ensure they all have schema property, just like createUiDef

Reply:
done

Reply:
Not required, since we check for the entire schema value string above

Reply:
how do we know that this file is a template file?  and not just a json file?

Reply:
In util.js, we have function to get all template files, only the json files schema match "schema.management.azure.com/schemas/(.*)/deploymentTemplate.json", we consider the json files as template files.

Reply:
That won't catch the case where the schema is missing...  We'll need to use something else, for example, the top level properties that are required (I think resources it the only one).